tests/testthat/test-shinytest2-adtteSpec.R

```diff
@@ -0,0 +1,226 @@
+        colData = data.frame(USUBJID = c("A", "B")) # USUBJID on experiment.
+      )
+    ),
+    colData = data.frame(
+      USUBJID = c("C", "D"), # USUBJID on MAE.
+      row.names = c("A", "B")
+    )
+  )
+  adtte <- data.frame(USUBJID = c("A", "B"))
+
+  expect_error(
+    h_km_mae_to_adtte(
+      adtte,
+      mae,
+      genes = hermes::gene_spec("ENSG1"),
+      experiment_name = "a"
+    ),
+    "Must be a subset of {'C','D'}, but has additional elements {'A','B'}",
+    fixed = TRUE
+  )
+})
+
+test_that("h_km_mae_to_adtte fails as expected if USUBJID in experiment colData is different from sample map", {
+  mae <- MultiAssayExperiment::MultiAssayExperiment(
+    experiments = list(
+      a = hermes::HermesDataFromMatrix(
+        matrix(1:4, 2, 2, dimnames = list(c("ENSG1", "ENSG2"), c("E", "F"))),
+        colData = data.frame(USUBJID = c("A", "B")) # USUBJID on experiment.
+      )
+    ),
+    colData = data.frame(
+      USUBJID = c("A", "B"),
+      row.names = c("E", "F") # USUBJID on sample map.
+    )
+  )
+  adtte <- data.frame(USUBJID = c("E", "F"))
+
+  expect_error(
+    h_km_mae_to_adtte(
+      adtte,
+      mae,
+      genes = hermes::gene_spec("ENSG1"),
+      experiment_name = "a"
+    ),
+    "Must be a subset of {'E','F'}, but has additional elements {'A','B'}",
+    fixed = TRUE
+  )
+})
+
+# adtteSpecInput ----
+
+test_that("adtteSpecInput creates expected HTML", {
+  expect_snapshot(adtteSpecInput(
+    "adtte",
+    label_paramcd = "Select right PARAMCD"
+  ))
+})
+
+test_that("{shinytest2} recording: adtteSpecServer module works as expected in the test app", {
+  skip_if_covr()
+  skip_if_too_deep(5)
+
+  app <- AppDriver$new(
+    app_dir = "adtteSpec",
+    name = "adtteSpecServer module works as expected in the test app"
+  )
+  ns <- module_ns_shiny2(app)
+  ns2 <- NS("teal-main_ui-filter_panel")
+
+  # check initialization
+  res <- app$get_values()
+  expect_equal(res$input[[ns("experiment-name")]], "hd1")
+  expect_equal(res$output[[ns("summary")]]$message, "please select at least one gene")
+
+  # check correct message
+  app$set_inputs(!!ns("genes-genes") := "GeneID:28")
+  res <- app$get_value(output = ns("summary"))
+  expect_equal(res$message, "please select an endpoint")
+
+  app$set_inputs(!!ns("adtte-paramcd") := "CRSD")
+  res <- app$get_value(output = ns("summary"))
+  expect_match(res, "CRSD")
+
+  app$set_inputs(!!ns("adtte-paramcd") := "PFS")
+  res <- app$get_value(output = ns("summary"))
+  expect_match(res, "PFS")
+
+  res <- app$get_value(input = ns("adtte-paramcd"))
+
+  # Test what happens if selected endpoint (here PFS) is no longer in filtered data.
+  app$set_inputs(!!ns2("add_ADTTE_filter-filter-var_to_add") := "PARAMCD")
+  app$set_inputs(!!ns2("ADTTE_filter-filter-_var_PARAMCD-content-selection") := "OS")
+
+  app$wait_for_idle()
+  # We expect to get a validation message (also a notification box but we cannot test that)
+  res <- app$wait_for_value(output = ns("summary"))
```

Comment:
```suggestion
  res <- app$get_value(output = ns("summary"))
```
? if it works

Reply:
the pattern `app$wait_for_idle()` followed by `app$wait_for_value(...)` has been eliminated